flang/lib/Optimizer/Passes/Pipelines.cpp

```diff
@@ -274,8 +274,10 @@ void createHLFIRToFIRPassPipeline(mlir::PassManager &pm, bool enableOpenMP,
     addNestedPassToAllTopLevelOperations<PassConstructor>(
         pm, hlfir::createInlineHLFIRAssign);
   pm.addPass(hlfir::createConvertHLFIRtoFIR());
-  if (enableOpenMP)
+  if (enableOpenMP) {
     pm.addPass(flangomp::createLowerWorkshare());
+    pm.addPass(flangomp::createLowerNontemporalPass());
```

Comment:
I think it would be better to include this pass much later in the pipeline so that loads and stores created by later passes are also included. Maybe around when the function attribute pass runs (right before FIR is lowered to LLVM).

There are also loads/stores created only during CodeGen (FIR to LLVM), but these are for boxes and so far as I can tell nontemporal only refers to data.